lib/src/local_working_copy.rs

```diff
@@ -1392,8 +1397,13 @@ impl TreeState {
                 MaterializedTreeValue::Tree(_) => {
                     panic!("unexpected tree entry in diff at {path:?}");
                 }
-                MaterializedTreeValue::Conflict { id: _, contents } => {
-                    self.write_conflict(&disk_path, contents)?
+                MaterializedTreeValue::Conflict { id, contents } => {
+                    let executable = if let Some(merge) = id.to_executable_merge() {
+                        merge.resolve_trivial().copied().unwrap_or_default()
+                    } else {
+                        false
+                    };
```

Comment:
I think it would make sense to make the materialized executable bit part of `MaterializedTreeValue::Conflict`. That would also make the "mode" field in `jj diff --git` a bit more accurate. Feel free to leave that for later.